uefi-mkconfig

```diff
@@ -53,156 +25,86 @@ add_uefi_entries () (
-				adding_kernel_commands="${adding_kernel_commands} initrd=${microcode_path//\//\\}intel-uc.img"
-			fi
-
-			# Check if corresponding initramfs exists
-			if [[ -f "$partition_mount$initramfs_image" ]]; then
-				adding_kernel_commands="${adding_kernel_commands} initrd=${initramfs_image//\//\\}"
-			else
-				ewarn "No initramfs found for \"$partition_mount$efi_file_path\"."
-			fi
-
-			# Add new entry
-			efibootmgr --create -b "$bootnum" --disk /dev/"$partition" --label "$entry_label" --loader "${efi_file_path//\//\\}"\
-			-u "$adding_kernel_commands" &>/dev/null || die "Failed to add UEFI entry for \"$efi_file_path\""
-			
-		else
-		
-			einfo "Existing UEFI Entry \"$uefi_entry_hex\" for \"$partition_mount$efi_file_path\" from partition \"$partition\" has been found."
+		done
 		
+		# Convert chosen entry ID into hex
+		bootnum="$(printf %04X $bootnum)"
+
+		# Get kernel version
+		## Remove everything before /
+		local kernel_version
+		kernel_version="${efi_file_path//*\//}"
+		## Remove everything before first - to remove any prefixes
+		kernel_version="${kernel_version/${kernel_version%%-*}-/}"
+		## Remove .efi suffix if it exists
+		if [[ "$kernel_version" == *".efi"* ]]; then
+			kernel_version="${kernel_version/\.${kernel_version##*.}}"
 		fi
-	done
-)
 
-remove_uefi_entries () {
-	local IFS=$'\n'		
-	for entry in $(efibootmgr -u | grep "$partition_partuuid"); do
-
-		# Get Hex number of a entry
-		local uefi_entry_hex
-		uefi_entry_hex="${entry:4:4}"
-		
-		# Check if the entry is within range managed by uefi-mkconfig
-		if [[ $((16#$uefi_entry_hex)) -gt 255 ]] && [[ $((16#$uefi_entry_hex)) -lt 513 ]]; then
-
-			local entry_efi_path
-			local entry_kernel_commands
-
-			# Create path to efi file
-			## Decide if entry is shim entry or not
-			if [[ "$entry" ==  *"File("*"shim"*")"* ]] || [[ "$entry" ==  *"File("*"SHIM"*")"* ]]; then
-				### Remove everything after and included with last character )
-				entry_efi_path="${entry##*(\.}"
-				### Remove everything after and included with last space
-				entry_efi_path="${entry_efi_path%%\ *}"
-				
-				### Prepare Entry kernel commands in case of shim booting
-				entry_kernel_commands=${entry##*\.efi}
-				entry_kernel_commands=${entry_kernel_commands//)*}
-				entry_kernel_commands=${entry_kernel_commands/#\ }
-			else
-				### Remove everything before first mention of string File(
-				entry_efi_path="${entry/${entry%%File(*}File\(/}"
-				### Remove everything after first ) character
-				entry_efi_path="${entry_efi_path%%)*}"
-
-				### Prepare Entry kernel commands
-				entry_kernel_commands=${entry/#*\.efi)}
-			fi
+		# Create label for UEFI eintry
+		local partition_label
+		local entry_label
+		partition_label="$(lsblk "/dev/$partition" -lno PARTLABEL)"
+		if [[ -n ${partition_label} ]]; then
+			entry_label="$kernel_version-$partition_label"
```

Comment:
Unrelated to this, but currently the label looks like this for me:
```
6.9.6-gentoo-dist-old-EFI System Partition
```
Aesthetically, I think that looks a bit odd, and I think it might be nicer to replace the hyphen with " on ", i.e.
```
6.9.6-gentoo-dist-old on EFI System Partition
```

Reply:
Indeed it looks better